test/views/errors/SubmissionErrorViewSpec.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.errors
+
+import assets.messages.SubmissionErrorMessages
+import auth.AuthKeys
+import models.auth.User
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.FakeRequest
+import views.ViewBaseSpec
+
+class SubmissionErrorViewSpec extends ViewBaseSpec {
+
+  "Rendering the Submission error page" when {
+
+    object Selectors {
+      val pageHeading = "#content h1"
+      val p1 = "#content p:nth-of-type(1)"
+      val p2 = "#content p:nth-of-type(2)"
+      val userLink = "#content a"
+    }
+
+    "the user is a non agent" should {
+
+      val user = User[AnyContentAsEmpty.type]("999999999")(fakeRequest)
+      lazy val view = views.html.errors.submission_error()(fakeRequest, mockAppConfig, messages, user)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      "have the correct document title" in {
+        document.title shouldBe SubmissionErrorMessages.title
+      }
+
+      "have a the correct page heading" in {
+        elementText(Selectors.pageHeading) shouldBe SubmissionErrorMessages.heading
+      }
+
+      "have the correct first paragraph on the page" in {
+        elementText(Selectors.p1) shouldBe SubmissionErrorMessages.p1
+      }
+
+      "have the correct second paragraph on the page" in {
+        elementText(Selectors.p2) shouldBe SubmissionErrorMessages.p2
+      }
+
+      "have a link to the vat-summary page" in {
+        element(Selectors.userLink).attr("href") shouldBe mockAppConfig.vatSummaryUrl
+      }
+    }
+
+    "the user is an agent" should {
+
+      lazy val fakeRequestWithClientsVRN: FakeRequest[AnyContentAsEmpty.type] =
+        FakeRequest().withSession(AuthKeys.agentSessionVrn -> "999999999")
+      lazy val agentUser: User[AnyContentAsEmpty.type] = User[AnyContentAsEmpty.type]("999999999", Some("XAIT012345678"))(fakeRequestWithClientsVRN)
+      lazy val view = views.html.errors.submission_error()(fakeRequestWithClientsVRN, mockAppConfig, messages, user = agentUser)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      "have the correct document title" in {
```

Comment:
With the content being the same apart from the link and link content I'd say you don't need all of these tests. Bit of an overkill